Tasks/ANTV1/anttask.ts

```diff
@@ -23,26 +25,26 @@ function pathExistsAsFile(path: string) {
 function publishTestResults(publishJUnitResults, testResultsFiles: string) {
     if (publishJUnitResults == 'true') {
         //check for pattern in testResultsFiles
+        let matchingTestResultsFiles;
         if (testResultsFiles.indexOf('*') >= 0 || testResultsFiles.indexOf('?') >= 0) {
             tl.debug('Pattern found in testResultsFiles parameter');
-            var buildFolder = tl.getVariable('System.DefaultWorkingDirectory');
-            var allFiles = tl.find(buildFolder);
-            var matchingTestResultsFiles = tl.match(allFiles, testResultsFiles, { matchBase: true });
+            const buildFolder = tl.getVariable('System.DefaultWorkingDirectory');
+            matchingTestResultsFiles = tl.findMatch(buildFolder, testResultsFiles, null, { matchBase: true });
         }
         else {
             tl.debug('No pattern found in testResultsFiles parameter');
-            var matchingTestResultsFiles = [testResultsFiles];
+            matchingTestResultsFiles = [testResultsFiles];
         }
 
-        if (!matchingTestResultsFiles || matchingTestResultsFiles.length == 0) {
+        if (!matchingTestResultsFiles || matchingTestResultsFiles.length === 0) {
             tl.warning('No test result files matching ' + testResultsFiles + ' were found, so publishing JUnit test results is being skipped.');
             return 0;
         }
 
-        var tp = new tl.TestPublisher("JUnit");
+        let tp = new tl.TestPublisher("JUnit");
```

Comment:
const?

Reply:
There is more cleanup needed in this file. I left several as "var" since they are being used as "global variables". I am going to leave this as is since I already tested it and I am seeing the telemetry in kusto.